crates/amaru-ledger/src/summary/rewards.rs

```diff
@@ -166,10 +166,16 @@ impl PoolState {
     ///
     /// The results is then used to calculate the _actual rewards_ based on the pool
     /// performances and its actual saturation level.
-    pub fn optimal_rewards(&self, available_rewards: Lovelace, total_stake: Lovelace) -> Lovelace {
+    pub fn optimal_rewards(
+        &self,
+        available_rewards: Lovelace,
+        total_stake: Lovelace,
+        protocol_parameters: &ProtocolParameters,
+    ) -> Lovelace {
         let one = SafeRatio::one();
-        let a0 = &*PLEDGE_INFLUENCE;
-        let z0 = safe_ratio(1, OPTIMAL_STAKE_POOLS_COUNT as u64);
+        let a0 = protocol_parameters.pledge_influence.clone();
+        let a0 = safe_ratio(a0.numerator, a0.denominator);
```

Comment:
That `.clone()` seems unnecessary. But the real problem is probably that don't use BigUInt ratio as types in the protocol parameter struct. So we have this conversion going on here for every pool. Now that we have a proper struct, I think we should simply parse and store them as BigUInt ratio when relevant (mostly all economic parameters needed in the rewards calculations).